### ISSUE TITLE
Link.Add should verify deserialisation of arrays before writing to file

### ISSUE THREAD

**Issue by mkromberg** [Dyalog/link#205]
**Describe the bug**
Serialisation and deserialisation of some arrays fails. I suggest that Link.Add test the ability to roudtrip before writing a file, to more clearly indentify the problem.

**To Reproduce**
```
      ]link.create something
      all←⍉(9⍴3)⊤⍳3*9
      ]link.add all
Added: #.all
Link Warning: ⎕SE.Link.Notify: created c:/tmp/myns/all.apla: invalid name defined by file: c:/tmp/myns/all.apla

```

**Expected behaviour**
Ideally, successful rountripping of all arrays, but failing that:
```
      ]link.add all
Unable to successfully convert array to array notation and back: LIMIT ERROR
```
Or in the case where there is no error, but the result does not match:

```
      ]link.add all
Unable to successfully convert array to array notation and back: array is modified by round trip
```

**Desktop (please complete the following information):**
 - `⎕SE.Link.Version`:  2.1.0-beta27

**Comment by mkromberg** [comment]
Secondarily, Link.Notify should not report a failure in Deserialise as "invalid name defined by file". So if the error is caused by external editing of an .apla file resulting in a failure to deserialise, I think we should say:

Link Warning: ⎕SE.Link.Notify: created c:/tmp/myns/all.apla: unable to deserialise array

**Comment by nicolas-dyalog** [comment]
Fixed in Link v2.1.0-beta34
